game/atmosphere.rpy

```diff
@@ -9,87 +9,181 @@ image dim light = "mod_assets/backgrounds/classroom/dim_light.png"
+        weather = random.choice(__RANDOM_WEATHER_TYPES)
 
         # Show the sky
-        renpy.show(name=sky, zorder=_SKY_Z_INDEX)
-        renpy.with_statement(trans=store.weather_change_transition)
+        renpy.show(name=weather.day_image, zorder=_SKY_Z_INDEX)
+        if with_transition:
+            renpy.with_statement(trans=store.weather_change_transition)
 
-        # Add the dimming effect matching the sky, if it exists
-        if dim:
-            renpy.show(name=dim, zorder=_DIM_Z_INDEX)
-            renpy.with_statement(trans=store.dim_change_transition)
+        # Add the dimming effect, if defined
+        if weather.dim_image:
+            renpy.show(name=weather.dim_image, zorder=_DIM_Z_INDEX)
+            if with_transition:
+                renpy.with_statement(trans=store.dim_change_transition)
 
         # Play rain sfx if the chosen weather is rainy
-        if weather_type in (JNWeatherTypes.rain, JNWeatherTypes.thunder):
-            renpy.music.play(filenames=__MUFFLED_RAIN_PATH, channel="weather_loop", fadein=3.0)
+        if weather.weather_sfx:
+            renpy.music.play(filenames=weather.weather_sfx, channel="weather_loop", fadein=3.0)
+
+        else:
+            renpy.music.stop(channel="weather_loop", fadeout=5.0)
 
         global current_weather
-        current_weather = weather_type
+        current_weather = weather.weather_type
 
-    def show_sky(weather_type):
+    def show_sky(weather, with_transition=True):
         """
         Shows the specified sky placeholder with associated dimming effect.
 
         IN:
-            weather_type - JNWeatherTypes value for the weather to set
+            weather - JNWeather to set
+            with_transition - If True, will visually fade in the new weather
         """
-        sky, dim = __WEATHER_EFFECT_MAP.get(weather_type)
-        if weather_type in (JNWeatherTypes.rain, JNWeatherTypes.thunder):
-            renpy.music.play(filenames=__MUFFLED_RAIN_PATH, channel="weather_loop", fadein=3.0)
-        
+        if weather.weather_sfx:
+            renpy.music.play(filenames=weather_sfx, channel="weather_loop", fadein=3.0)
+
         else:
             renpy.music.stop(channel="weather_loop", fadeout=5.0)
 
-        renpy.show(name=sky, zorder=_SKY_Z_INDEX)
-        renpy.with_statement(trans=store.weather_change_transition)
+        renpy.show(name=weather.day_image, zorder=_SKY_Z_INDEX)
+        if with_transition:
+            renpy.with_statement(trans=store.weather_change_transition)
 
-        if dim:
-            renpy.show(name=dim, zorder=_DIM_Z_INDEX)
-            renpy.with_statement(trans=store.dim_change_transition)
+        if weather.dim_image:
+            renpy.show(name=weather.dim_image, zorder=_DIM_Z_INDEX)
+            if with_transition:
+                renpy.with_statement(trans=store.dim_change_transition)
 
         global current_weather
-        current_weather = weather_type
+        current_weather = weather.weather_type
```

Comment:
We can then store the current JNWeather object directly and switch to it using it's method or with a wrapper function.
That said, it's likely there will be a few changes to be made once I finally fix those libs >.> anyways so it might be for the best to leave it for now. I'm still curious what you think of this though.

Reply:
As above - I'm for it, but would prefer to tackle under a dedicated overhaul.

Reply:
Yeah I agree